pinot-core/src/main/java/com/linkedin/pinot/core/operator/transform/transformer/datetime/BaseDateTimeTransformer.java

```diff
@@ -62,7 +62,7 @@ protected long transformMillisToEpoch(long millisSinceEpoch) {
   }
 
   protected String transformMillisToSDF(long millisSinceEpoch) {
-    return _outputDateTimeFormatter.print(millisSinceEpoch);
+    return _outputDateTimeFormatter.print(transformToOutputGranularity(millisSinceEpoch));
```

Comment:
Should not embed **transformToOutputGranularity** into **transformMillisToSDF**, but add it into **Epoch/SDFToSDFTransformer** (refer to **SDFToEpochTransformer**)

Reply:
Ahh I see now thank you, I will get that changed.